frontend/src/metabase/dashboard/dashboard.js

```diff
@@ -152,7 +152,14 @@ export const fetchCardData = createThunkAction(FETCH_CARD_DATA, function(card, d
             }
         }, DATASET_SLOW_TIMEOUT);
 
-        result = await MetabaseApi.dataset(datasetQuery);
+
+        try {
+          result = await MetabaseApi.dataset(datasetQuery);
+        }
+        catch (error) {
+          clearTimeout(slowCardTimer);
+          return { dashcard_id: dashcard.id, card_id: card.id, result: { error } };
+        }
```

Comment:
You could replace both `clearTimeout(slowCardTimer)` instances with a `finally { clearTimeout(slowCardTimer); }`